### PR TITLE
[PREVIEW] Feature/sl 1471 history of changes listing

### PR DESCRIPTION
**Pull request by kpakur**
SL-1471

### CODE REVIEW

File: src/main/resources/db/schema/SL-1471-history-of-changes-listing.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                name: case_number
+                type: VARCHAR(200)
+            - column:
+                name: case_title
+                type: VARCHAR(200)
+            - column:
+                name: case_type
+                type: VARCHAR(100)
+            - column:
+                name: hearing_type
```
Comment (zielinskikamil) [inline src/main/resources/db/schema/SL-1471-history-of-changes-listing.yaml]: So far code looks good. I just noticed some inconsistency with US. In the requirements from SL-1349, we've had to add hearing_type table (with code and description) but there was no information about connecting hearing type to listing hearing part and here we have hearing_type column that is not connected with new table.

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/security/S2SJwtAuthenticationFilterTest.java [hunk 2]
```diff
@@ -12,78 +14,143 @@
...
-    public void validateToken_respondsWithError_forNoTokenInProperHeader() throws ServletException, IOException {
-        when(s2SAuthenticationService.validateToken(any())).thenReturn(false);
-        when(request.getHeader(S2SAuthenticationService.HEADER_NAME))
-            .thenReturn(S2SAuthenticationService.HEADER_CONTENT_PREFIX);
+    public void doFilterInternal_notNeededAuth_forOpenEndpoint() throws ServletException, IOException {
+        when(request.getHeader(TokenService.HEADER_NAME))
+            .thenReturn("NO BEARER IN HEADER");
         when(request.getRequestURI())
-            .thenReturn("/needingJwtUrl");
+            .thenReturn("/health"); //
```
Comment (zielinskikamil) [inline src/test/java/uk/gov/hmcts/reform/sandl/snlevents/security/S2SJwtAuthenticationFilterTest.java]: Can you remove this empty comment?

File: src/main/resources/db/schema/SL-1471-history-of-changes-listing.yaml (hunk 1 above)
Comment (kpakur) [reply src/main/resources/db/schema/SL-1471-history-of-changes-listing.yaml]: discussed

File: src/test/java/uk/gov/hmcts/reform/sandl/snlevents/security/S2SJwtAuthenticationFilterTest.java (hunk 2 above)
Comment (kpakur) [reply src/test/java/uk/gov/hmcts/reform/sandl/snlevents/security/S2SJwtAuthenticationFilterTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/config/H2DbExtendedDialect.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.sandl.snlevents.config;
+
+import org.hibernate.dialect.H2Dialect;
+
+import java.sql.Types;
+
+public class H2DbExtendedDialect extends H2Dialect {
```
[the remark this answers is not in the document]
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlevents/config/H2DbExtendedDialect.java]: good, done

**Comment by satyachundur** [comment]
All looking good, ran tests locally , however quick question i have seen no functional tests associated with this repo.  It has only smoke and health check @kpakur

**Pull request merged**